packages/presentation/lib/screen/main_screen.dart

```diff
@@ -0,0 +1,124 @@
+          ),
+        );
+
+  @override
+  void initState() {
+    super.initState();
+    bloc.dialogStream.listen((event) {
+      if (event is GameDialog) {
+        showDialog(
+          context: context,
+          builder: (context) {
+            return AlertDialog(
+              content: Text(
+                event.message,
+                style: const TextStyle(fontSize: 30),
+              ),
+            );
+          },
+        );
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return StreamBuilder(
+        stream: bloc.dataStream,
+        builder: (context, snapshot) {
+          if (snapshot.data != null) {
+            final data = snapshot.data as BlocData;
+            final blocData = data.data;
+            return Scaffold(
+              appBar: AppBar(
+                title: const Text('Guess the Number'),
+                centerTitle: true,
+              ),
+              body: SafeArea(
+                child: Center(
+                  child: Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      Center(
+                        child: Text(
+                          'Guess the number: ${blocData.randomNumber}',
+                          style: const TextStyle(fontSize: 28),
+                        ),
+                      ),
+                      const SizedBox(
+                        height: 30,
+                      ),
+                      Padding(
+                        padding: const EdgeInsets.fromLTRB(90, 0, 90, 0),
+                        child: TextField(
+                          controller: bloc.editController,
+                          decoration: const InputDecoration(
+                            border: OutlineInputBorder(),
+                            labelText: 'Your guess',
+                          ),
+                          keyboardType: TextInputType.number,
+                        ),
+                      ),
+                      const SizedBox(
+                        height: 30,
+                      ),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.center,
+                        children: [
+                          ElevatedButton(
+                            onPressed: blocData.isDisabled
+                                ? null
+                                : () {
+                                    bloc.checkNumber();
+                                  },
+                            child: const Text(
+                              'Try',
+                              style: TextStyle(fontSize: 20),
+                            ),
+                          ),
+                          const SizedBox(
+                            width: 30,
+                          ),
+                          ElevatedButton(
+                            onPressed: () {bloc.generateRandomNumber();},
```

Comment:
`onPressed: () {bloc.generateRandomNumber();},` -> `onPressed: bloc.generateRandomNumber,`

Reply:
It's breaking the game this way. I guess it's because this function is void, so it makes onPressed method think that we passed null to it and it's locking the button, making it inactive. And also it continuously calls this method bloc.generateRandomNumber, so that makes it generate numbers unstoppable.

Reply:
Or because passing just onPressed: bloc.generateRandomNumber passed the result of generateRandomNumber, but not the call of it